client/src/pages/course/submit-scores.tsx

```diff
@@ -164,80 +170,78 @@ export function SubmitScorePage() {
   );
 }
 
-async function parseFiles(incomingFiles: any) {
+async function parseFiles(incomingFiles: IncomingFiles): Promise<StudentScore[]> {
   const files = incomingFiles.fileList;
+
   const filesContent: string[] = await Promise.all(
     files.map(
-      (file: any) =>
-        new Promise((res, rej) => {
+      file =>
+        new Promise<string>((res, rej) => {
           const reader = new FileReader();
-          reader.readAsText(file.originFileObj, 'utf-8');
-          reader.onload = ({ target }) => res(target ? target.result : '');
+          reader.readAsText(file.originFileObj as Blob, 'utf-8');
+          reader.onload = ({ target }) => res((target?.result as string) || '');
           reader.onerror = e => rej(e);
         }),
     ),
   );
-  const scores = (await Promise.all(filesContent.map((content: string) => csv().fromString(content))))
-    .reduce((acc, cur) => acc.concat(cur), [])
-    .map(item => {
-      if (isUndefined(item.Github) || isUndefined(item.Score)) {
-        throw new Error('Incorrect data: CSV file should content the headers named "Github" and "Score"!');
-      }
-      return {
-        score: parseInt(item.Score, 10),
-        github: filterLogin(item.Github).toLowerCase(),
-      };
-    });
+
+  const parsedRecords = await Promise.all(filesContent.map(content => csv().fromString(content)));
+  const scores = parsedRecords.flat().map(item => {
+    if (isUndefined(item.Github) || isUndefined(item.Score)) {
+      throw new Error('Incorrect data: CSV file should contain the headers named "Github" and "Score"!');
+    }
+
+    const parsedScore = parseInt(item.Score, 10);
+    return {
+      score: isNaN(parsedScore) ? 0 : parsedScore,
```

Comment:
This is not correct. If we couldn't parse score, we shouldn't map it to `0`, it's better to throw error instead to force user fix csv